From cb1b94ffa91d71aba8e07ed018e9f43064e05a49 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Diego=20Elio=20'Flameeyes'=20Petten=C3=B2?= <flameeyes@gmail.com>
Date: Wed, 10 Jun 2009 17:24:02 +0200
Subject: [PATCH 3/3] Fix building with Linux kernel 2.6.30 and later.

With commit 99b76233803beab302123d243eea9e41149804f3 the proc_dir_entry
structure does not have an owner any longer, this just removes the
assignment to make the code build with the newer kernel.
---
 kernel/config.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/kernel/config.c b/kernel/config.c
index d6c4b5e..34681f5 100644
--- a/kernel/config.c
+++ b/kernel/config.c
@@ -5,6 +5,7 @@
  */
 
 #include <linux/proc_fs.h>
+#include <linux/version.h>
 
 #include "iscsi.h"
 #include "iscsi_dbg.h"
@@ -43,7 +44,9 @@ int iet_procfs_init(void)
 	if (!(proc_iet_dir = proc_mkdir("iet", init_net.proc_net)))
 		goto err;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 30)
 	proc_iet_dir->owner = THIS_MODULE;
+#endif
 
 	for (i = 0; i < ARRAY_SIZE(iet_proc_entries); i++) {
 		ent = create_proc_entry(iet_proc_entries[i].name, 0, proc_iet_dir);
-- 
1.6.3.1

